src/main/java/org/broadinstitute/hellbender/tools/dragstr/EstimateDragstrParameters.java

```diff
@@ -0,0 +1,935 @@
+    private String output = null;
+
+    @Argument(fullName= DEBUG_SITES_OUTPUT_FULL_NAME, doc = "table with information gather on the samples sites. Includes what sites were downsampled, disqualified or accepted for parameter estimation", optional = true)
+    private String sitesOutput = null;
+
+    private SAMSequenceDictionary dictionary;
+
+    @Override
+    public boolean requiresReference() {
+        return true;
+    }
+
+    @Override
+    public boolean requiresReads() {
+        return true;
+    }
+
+    @Override
+    protected void onStartup() {
+        super.onStartup();
+        hyperParameters.validate();
+        dictionary = getBestAvailableSequenceDictionary();
+        if (runInParallel) {
+            if (threads == 1) {
+                logger.warn("parallel processing was requested but the number of threads was set to 1");
+            }
+        } else if (threads > 1) {
+            runInParallel = true;
+        }
+        if (runInParallel) {
+            if (threads == 0) {
+                logger.info("Running in parallel using the system suggested default thread count: " + Runtime.getRuntime().availableProcessors());
+            } else {
+                logger.info("Running in parallel using the requested number of threads: " + threads);
+            }
+        }
+    }
+
+    @Override
+    public void traverse() {
+        hyperParameters.validate();
+        dictionary = getBestAvailableSequenceDictionary();
+        final List<SAMReadGroupRecord> readGroups = hasReads() ? getHeaderForReads().getReadGroups() : Collections.emptyList();
+        final List<String> readGroupIds = readGroups.stream()
+                .map(SAMReadGroupRecord::getId)
+                .collect(Collectors.toList());
+        final List<String> sampleNames = readGroups.stream()
+                .map(SAMReadGroupRecord::getSample)
+                .distinct().collect(Collectors.toList());
+        final Optional<String> sampleName = resolveSampleName(sampleNames);
+
+        try (final PrintWriter sitesOutputWriter = openSitesOutputWriter(sitesOutput);
+             final STRTableFile strTable = STRTableFile.open(strTablePath)) {
+
+            checkSequenceDictionaryCompatibility(dictionary, strTable.dictionary());
+            final StratifiedDragstrLocusCases allSites;
+            final List<SimpleInterval> intervals = getTraversalIntervals();
+
+            runInParallel |= threads > 1;
+            if (runInParallel) {
+                if (threads == 1) {
+                    logger.warn("parallel processing was requested but the number of threads was set to 1");
+                }
+                allSites = collectCaseStatsParallel(intervals, shardSize, strTable);
+            } else {
+                allSites = collectCaseStatsSequencial(intervals, strTable);
+            }
+            logSiteCounts(allSites, "all loci/cases");
```

Comment:
Yes, downsampling happens in two stages. Don't try to look to hard into it... there are e few things that don't make that much sense at least at first but this is the way Dragen does it. 
The first decimation occurs in the Compose STR Table tool (sample agnostic) and the objective is to end up with enough sites for every P, L combination so that for any given sample there should be enough informative sites once the second downsample is done. 

In the second downsampling we continue with the decimation were it was left off reducing even further based on the sample data this time. Technically there is no need for the original decimation table in the second phase as we can infer it based on the lowest bit set in any mask for that P,L combo, however since it is part of the metadat in STR Table zip we use it so that we don't need to infer it.
In t